src/Microsoft.TemplateEngine.Edge/Settings/SettingsFilePaths.cs

```diff
@@ -0,0 +1,233 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.TemplateEngine.Abstractions;
+
+namespace Microsoft.TemplateEngine.Edge.Settings
+{
+    internal class SettingsFilePaths
+    {
+        private readonly IEngineEnvironmentSettings _environmentSettings;
+        private string? _aliasesFile;
+        private string? _firstRunCookie;
+        private string? _scratchDir;
+        private string? _settingsFile;
+        private string? _contentDir;
+        private string? _templatesCacheFile;
+
+        internal SettingsFilePaths(IEngineEnvironmentSettings environmentSettings)
+        {
+            _environmentSettings = environmentSettings;
+        }
+
+        internal string AliasesFile => GetOrComputePath(ref _aliasesFile, BaseDir, "aliases.json");
+
+        internal string BaseDir => _environmentSettings.Paths.HostVersionSettingsDir;
+
+        internal string Content => GetOrComputePath(ref _contentDir, BaseDir, "content");
+
+        internal string FirstRunCookie => GetOrComputePath(ref _firstRunCookie, BaseDir, ".firstrun");
+
+        internal string ScratchDir => GetOrComputePath(ref _scratchDir, BaseDir, "scratch");
+
+        internal string SettingsFile => GetOrComputePath(ref _settingsFile, BaseDir, "settings.json");
+
+        internal string TemplateCacheFile => GetOrComputePath(ref _templatesCacheFile, BaseDir, "templatecache.json");
+
+        internal string ProcessPath(string path)
+        {
+            if (string.IsNullOrEmpty(path))
+            {
+                return path;
+            }
+
+            if (path[0] != '~')
+            {
+                return path;
+            }
+
+            return Path.Combine(_environmentSettings.Paths.UserProfileDir, path.Substring(1));
+        }
+
+        internal void Copy(string path, string targetPath)
+        {
+            if (_environmentSettings.Host.FileSystem.FileExists(path))
+            {
+                _environmentSettings.Host.FileSystem.FileCopy(path, targetPath, true);
+                return;
+            }
+
+            foreach (string p in EnumerateFiles(path, "*", SearchOption.AllDirectories).OrderBy(x => x.Length))
+            {
+                string localPath = p.Substring(path.Length).TrimStart('\\', '/');
```

Comment:
Another very specific method for Scanner. Both are guaranteed to be absolute.
Honestly thinking about moving to Scanner as private...

Reply:
I decided not to move these methods for time being. I hope we will refactor them soon; however since they are internal now they are not subject of public API scope. I would prefer to keep them as are to avoid bugs with target to refactor them when refactoring Edge. Same applies to other comments in this file.